web-ui/controller/index.ml

```diff
@@ -1,21 +1,21 @@
 module Client = Ocaml_ci_api.Client
 
-let ( >>!= ) = Git_forge.( >>!= )
+let ( >>!!= ) = Git_forge.( >>!= )
+let ( >|!= ) a f = Lwt_result.map f a
 
-let list_orgs prefix ci =
+let list_orgs ~orgs =
   let open Lwt.Infix in
-  Backend.ci ci >>= Client.CI.orgs_detailed >>!= fun orgs ->
-  let orgs = List.filter (fun o -> o.Client.CI.number_repos > 0) orgs in
-  Dream.respond @@ View.Index.list_orgs prefix orgs
-
-let list_all_orgs ~github ~gitlab =
-  let open Lwt.Infix in
-  Backend.ci gitlab >>= Client.CI.orgs_detailed >>!= fun gitlab_orgs ->
-  Backend.ci github >>= Client.CI.orgs_detailed >>!= fun github_orgs ->
-  let github_orgs =
-    List.filter (fun o -> o.Client.CI.number_repos > 0) github_orgs
+  let extract ((prefix, name, backend) : string * string * Backend.t) =
+    Backend.ci backend
+    >>= Client.CI.orgs_detailed
+    >|!= List.filter (fun o -> o.Client.CI.number_repos > 0)
+    >|!= fun orgs -> (prefix, name, orgs)
   in
-  let gitlab_orgs =
-    List.filter (fun o -> o.Client.CI.number_repos > 0) gitlab_orgs
+  let to_result_list orgs org =
+    match org with
+    | Ok org -> Result.map (fun orgs -> org :: orgs) orgs
+    | Error _ -> orgs
```

Comment:
It should be possible to also use `Result.bind` on `org` if we're starting to use `Result` combinators.